app/scripts/modules/kubernetes/serverGroup/configure/wizard/Clone.controller.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.serverGroup.configure.kubernetes.clone', [
+  require('angular-ui-router'),
+  require('../../../../core/application/modal/platformHealthOverride.directive.js'),
```

Comment:
When an Angular app initializes, it finds all its dependencies starting with the `ng-app` declaration and builds up a list of everything it needs, then initializes it. It's not really concerned with what a particular module declares as its dependencies - if it finds a match for something it's injecting based on its name, it will use it, even if it's not explicitly declared in that component.

Basically, all those components are `require`d somewhere in the application, so it knows what to inject.

The problem comes up when you try to write a test for that component. You don't want Angular to initialize the entire application for each individual test, because that would be very slow - especially in an application of this size. So you tell it to just load the component you want to test. If some of the components it needs to inject aren't declared as dependencies, it'll throw an exception and not run the test.